Dockerfile

```diff
@@ -92,134 +109,97 @@ ARG VERSION_PKG="github.com/talos-systems/talos/internal/pkg/version"
-COPY --from=init-build /init /init
-
-ARG TOOLCHAIN_IMAGE
-FROM ${TOOLCHAIN_IMAGE} AS initramfs-archive
+FROM scratch AS kernel
+COPY --from=docker.io/autonomy/kernel:2ac99a0 /boot/vmlinuz /vmlinuz
+
+# The initramfs target provides the Talos initramfs image.
+
+FROM scratch AS initramfs-base
+COPY --from=docker.io/autonomy/fhs:8467184 / /
+COPY --from=docker.io/autonomy/ca-certificates:20f39f7 / /
+COPY --from=docker.io/autonomy/dosfstools:767dee6 / /
+COPY --from=docker.io/autonomy/musl:9bc7430 / /
+COPY --from=docker.io/autonomy/syslinux:85e1f9c / /
+COPY --from=docker.io/autonomy/xfsprogs:5e50579 / /
+FROM initramfs-base AS initramfs-build
+COPY --from=init /init /init
+FROM build AS initramfs-archive
 COPY --from=initramfs-build / /initramfs
 WORKDIR /initramfs
 RUN set -o pipefail && find . 2>/dev/null | cpio -H newc -o | xz -v -C crc32 -0 -e -T 0 -z >/initramfs.xz
-
 FROM scratch AS initramfs
 COPY --from=initramfs-archive /initramfs.xz /initramfs.xz
 
-# The rootfs target creates the root filesystem archive.
-
-ARG ROOTFS_IMAGE
-FROM ${ROOTFS_IMAGE} AS rootfs-build
-COPY --from=kernel /modules /
-COPY images /usr/images
-
-ARG TOOLCHAIN_IMAGE
-FROM ${TOOLCHAIN_IMAGE} AS rootfs-archive
-COPY --from=rootfs-build / /rootfs
+# The rootfs target provides the Talos rootfs image.
+
+FROM tools AS rootfs-build
+COPY --from=docker.io/autonomy/fhs:8467184 / /rootfs
+COPY --from=docker.io/autonomy/ca-certificates:20f39f7 / /rootfs
+COPY --from=docker.io/autonomy/containerd:03821f9 / /rootfs
+COPY --from=docker.io/autonomy/cni:063e06f / /rootfs
+COPY --from=docker.io/autonomy/dosfstools:767dee6 / /rootfs
+COPY --from=docker.io/autonomy/eudev:05186a8 / /rootfs
+COPY --from=docker.io/autonomy/iptables:a7aa58f / /rootfs
+COPY --from=docker.io/autonomy/libressl:3fca2cf / /rootfs
+COPY --from=docker.io/autonomy/libseccomp:80ea634 / /rootfs
+COPY --from=docker.io/autonomy/musl:9bc7430 / /rootfs
+COPY --from=docker.io/autonomy/runc:c79f79d / /rootfs
+COPY --from=docker.io/autonomy/socat:032c783 / /rootfs
+COPY --from=docker.io/autonomy/syslinux:85e1f9c / /rootfs
+COPY --from=docker.io/autonomy/xfsprogs:5e50579 / /rootfs
+COPY --from=docker.io/autonomy/images:150048d / /rootfs
+COPY --from=docker.io/autonomy/kubeadm:8607389 / /rootfs
+COPY --from=docker.io/autonomy/crictl:ddbeea1 / /rootfs
+COPY --from=docker.io/autonomy/base:f9a4941 /toolchain/lib/libblkid.* /rootfs/lib
+COPY --from=docker.io/autonomy/base:f9a4941 /toolchain/lib/libuuid.* /rootfs/lib
+COPY --from=docker.io/autonomy/base:f9a4941 /toolchain/lib/libkmod.* /rootfs/lib
+COPY --from=docker.io/autonomy/kernel:2ac99a0 /lib/modules /rootfs/lib/modules
+COPY images/*.tar /rootfs/usr/images
+COPY ./hack/cleanup.sh /toolchain/bin/cleanup.sh
+RUN ["/toolchain/bin/cleanup.sh", "/rootfs"]
+FROM scratch AS rootfs-base
+COPY --from=rootfs-build /rootfs /
+FROM build AS rootfs-archive
+COPY --from=rootfs-base / /rootfs
 WORKDIR /rootfs
-RUN tar -cvpzf /rootfs.tar.gz .
-
+RUN tar -cpzf /rootfs.tar.gz .
```

Comment:
I remove the verbose flags because before I was running `cleanup.sh` there were a lot of source code files and I realized that printing out each one when tarring was slowing down builds dramatically.